src/workerd/api/sockets.c++

```diff
@@ -9,8 +9,47 @@
 namespace workerd::api {
 
 
+bool isValidAddress(kj::StringPtr address) {
+  // This function performs some basic length and characters checks, it does not guarantee that
+  // the specified address is a valid domain. It should only be used to reject malicious
+  // addresses.
+  if (address.size() > (255 + 6)) {
+    // RFC1035 states that maximum domain name length is 255 octets. But we also add 6 to support
+    // port numbers in the address.
+    //
+    // IP addresses are always shorter, so we take the max domain length instead.
+    return false;
+  }
+
+  for (int i = 0; i < address.size(); i++) {
+    switch (address[i]) {
+      case '-':
+      case '.':
+      case ':':
+      case '_':
+      case '[': case ']': // For IPv6.
+      case '%': // For percent-encoded domain names.
+      case '/': // To enable "unix:/" addresses (only used for testing, the proxy should reject this).
```

Comment:
Can we change the tests to not rely on this?

This is actually something I had made a note to myself to bring up. It looks like you've written the tests such that `connect()` is backed by a raw network `HttpClient`, and then you're setting up unix sockets which the test connects to through this. Instead, could you use a mock `HttpClient`? Then the test can directly check that the hostname is as expected and return any kind of connection it wants (e.g. one implemented in-process) without relying on real sockets. This would make the test more hermetic (doesn't touch the filesystem) and would allow us to ban unix socket addresses here which would be reassuring.

Reply:
I can do that, but let's please not let it block this PR. I will make a note to come back and disallow `/` in the address (EW-7144).

Reply:
Actually thinking about it some more, don't we want to allow workerd users to connect to unix sockets?

Reply:
> let's please not let it block this PR.

OK but I'd like to see slashes disallowed before the API becomes publicly available.

> Actually thinking about it some more, don't we want to allow workerd users to connect to unix sockets?

I was thinking about it, and I think probably not.

Certainly workerd instances may want to define _specific_ unix sockets that represent backends they want to connect to, which they can do today by defining an `ExternalServer` in the config file specific to each one. The question here is whether worker code should be able to open arbitrary paths. I think the answer is probably "no". Giving a Worker the ability to open arbitrary paths probably defeats all sandboxing and also makes the worker tightly coupled to the host system's file layout.